### PR TITLE
fix: update in-repo protobufs for roster, HIP-904 codes, new state changes

### LINKED ISSUE #14920: Update protobufs for remaining state changes
### Problem

We need to update the in-repo protobufs to reflect,
 - The roster protobufs (c.f. [here](https://github.com/hashgraph/hedera-protobufs/commit/660284c6f64812eeed6b677a163bb464c989aaa8)).
 - The new response codes for HIP-904 (c.f. [here](https://github.com/hashgraph/hedera-protobufs/commit/15b6a5b85f47bd3aaeccb952c30064e2cc2a2dba))
 - New `StateChanges` enums to support consolidated transaction receipts and `PlatformState` changes.

### Solution

Do the above.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #14920

### CODE REVIEW

**Review by kfa-aguda — APPROVED** [review]
roster changes look good as replicated from hedera-protobuf.

File: hapi/hedera-protobufs/block/stream/output/state_changes.proto [hunk 1]
```diff
@@ -786,17 +797,9 @@ message QueuePushChange {
...
-         * Do we really intend to push `transaction records` as state
-         * changes in the block stream? I would think that any queue holding
-         * transaction records is temporary storage (e.g. record cache) and
-         * will be removed or replaced with a block item queue as part
-         * of the change over to block stream.
-         * </blockquote></blockquote>
+         * All the transaction receipts for a round added to queue state.
          */
-        proto.TransactionRecordEntry transaction_record_entry_element = 3;
+        proto.TransactionReceiptEntries transaction_receipt_entries_element = 3;
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/block/stream/output/state_changes.proto:802]: The removed question is relevant.
Do we actually need these as state changes?  Can we elide this list and save the bytes in the block stream, or is it something that is actually required in state to maintain deterministic behavior?

File: hapi/hedera-protobufs/services/response_code.proto [hunk 2]
```diff
@@ -1562,4 +1562,27 @@ enum ResponseCodeEnum {
      * this transaction.
      */
     ACCOUNT_HAS_PENDING_AIRDROPS = 365;
+
+    /**
+     * Consensus throttle did not allow execution of this transaction. The transaction is throttled at
+     * consensus level.
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/response_code.proto:1568]: Second sentence basically repeats the first.
```suggestion
     * Consensus throttle did not allow execution of this transaction.<br/>
     * The transaction should be retried after a modest delay.
```

File: hapi/hedera-protobufs/services/response_code.proto [hunk 3]
```diff
@@ -1562,4 +1562,27 @@ enum ResponseCodeEnum {
...
+     */
+    THROTTLED_AT_CONSENSUS = 366;
+
+    /**
+     * The provided pending airdrop id is invalid. It could be that the pending airdrop is already claimed
+     * or cancelled.
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/response_code.proto:1574]: Could provide more guidance for resolution:
```suggestion
     * The provided pending airdrop id is invalid.<br/>
     * This pending airdrop MAY already be claimed or cancelled.
     * <p>
     * The client SHOULD query a mirror node to determine the current status of
     * the pending airdrop.
```

File: hapi/hedera-protobufs/services/response_code.proto [hunk 4]
```diff
@@ -1562,4 +1562,27 @@ enum ResponseCodeEnum {
...
+     */
+    INVALID_PENDING_AIRDROP_ID = 367;
+
+    /**
+     * The token having fallback royalty fee cannot be airdropped or claimed.
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/response_code.proto:1579]: Wording is a bit awkward:
```suggestion
     * The token to be airdropped has a fallback royalty fee and cannot be
     * sent or claimed via an airdrop transaction.
```

File: hapi/hedera-protobufs/services/response_code.proto [hunk 5]
```diff
@@ -1562,4 +1562,27 @@ enum ResponseCodeEnum {
...
+     */
+    TOKEN_AIRDROP_WITH_FALLBACK_ROYALTY = 368;
+
+    /**
+     * The token in the pending airdrop that was tried to claim is deleted or the sender is no more
+     * owning the token.
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/response_code.proto:1585]: Very awkward wording here.
```suggestion
     * This airdrop claim is for a pending airdrop with an invalid token.<br/>
     * The token might be deleted, or the sender may not have enough tokens
     * to fulfill the offer.
     * <p>
     * The client SHOULD query mirror node to determine the status of the pending
     * airdrop and whether the sender can fulfill the offer.
```

File: hapi/hedera-protobufs/services/state/roster/roster.proto
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/state/roster/roster.proto:1]: Should this be part of the platform state (in `hapi/hedera-protobufs/platform`)?

**Review by jsync-swirlds — COMMENTED** [review]
A few wording updates, and a couple questions.

File: hapi/hedera-protobufs/block/stream/output/state_changes.proto (hunk 1 above)
Comment (tinker-michaelj) [reply hapi/hedera-protobufs/block/stream/output/state_changes.proto:802]: We need the last 180 consensus seconds of receipts in state for deterministic duplicate classification and fee charging.

There may be future optimizations to maintain this information out of state; but I think for now we need it.

File: hapi/hedera-protobufs/services/state/roster/roster.proto
Comment (tinker-michaelj) [reply hapi/hedera-protobufs/services/state/roster/roster.proto:1]: Hmm not sure, I just used the same path as [here](https://github.com/hashgraph/hedera-protobufs/blob/main/services/state/roster/roster.proto). I think TSS is now an application service so probably intentional?

**Review by jsync-swirlds — APPROVED** [review]

**Pull request merged**